features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Applications/Create/NewApp.jsx

```diff
@@ -253,13 +258,57 @@ class NewApp extends React.Component {
         return Promise.resolve(true);
     };
 
+    /**
+     * add a new group function
+     * @param {*} chip newly added group
+     * @param {*} appGroups already existing groups
+     */
+    handleAddChip = (chip, appGroups) => {
+        this.setState(() => {
```

Comment:
@CrowleyRajapakse  `setState` callback is not required unless you are using current state to set the next state

Reply:
resolved in the latest code